### ISSUE TITLE
C error: expected expression before ‘Option_array_string’

### ISSUE THREAD

**Issue by eigenlambda** [vlang/v#3058]
**V version:** V 0.1.23 e182274
**OS:** Ubuntu 18.04 in WSL 1 on Windows 10

**What did you do?**
```
>>> user@User-PC:~$ v -g
V 0.1.23 e182274
Use Ctrl-C or `exit` to exit
>>> import os
>>> for x in os.ls('.') or {[]string} {println(x)}
```

**What did you expect to see?**
the list of things in .
 
**What did you see instead?**
```
/tmp/v/.vrepl_temp.tmp.c:4669:22: error: expected expression before ‘Option_array_string’
  array_string tmp1 = Option_array_string tmp2 =  os__ls ( tos3(".") ) ;
                      ^~~~~~~~~~~~~~~...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen.
Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

**Comment by medvednikov** [comment]
Right now you have to do `files := os.ls('.') or {[]string}` and `for file in files {...`

**Comment by spaceface777** [comment]
Closed in favor of #6395 which contains the same issue and is more up-to-date.
